napari/_vispy/layers/image.py

```diff
@@ -323,11 +323,12 @@ def downsample_texture(
     "r8": np.dtype(np.uint8),
     "r16": np.dtype(np.uint16),
     "r32f": np.dtype(np.float32),
-    None: np.dtype(np.float32),
 }
 
 _DTYPE_TO_VISPY_FORMAT = {v: k for k, v in _VISPY_FORMAT_TO_DTYPE.items()}
 
+_VISPY_FORMAT_TO_DTYPE[None] = np.dtype(np.float32)
+
```

Comment:
@jni, this is a real fix. 

As without this change the `_DTYPE_TO_VISPY_FORMAT[np.float32]` return `None`, not `"r32f"` here:

https://github.com/napari/napari/blob/8ed7f4c62fa607ed1f832b140043559b0b4683d9/napari/_vispy/layers/labels.py#L581

Reply:
please add a comment explaining this just above, it seems easy to overlook this in the future and move it back to the dict declaration.

Reply:
Ok now?